src/dotnet/commands/dotnet-publish/PublishCommand.cs

```diff
@@ -153,7 +155,13 @@ private bool PublishProjectContext(ProjectContext context, string buildBasePath,
             var exporter = context.CreateExporter(configuration);
 
             var isPortable = string.IsNullOrEmpty(context.RuntimeIdentifier);
-            foreach (var export in exporter.GetAllExports())
+
+            // Collect all exports and organize them
+            var exports = exporter.GetAllExports()
+                .ToDictionary(e => e.Library.Identity.Name);
```

Comment:
This could be dangerous as proven by lots of `Duplicated key` errors in `DepencyModel`